examples/ecs/relationships.rs

```diff
@@ -0,0 +1,192 @@
+        // through the use of atomic counters.
+        let alice = commands.spawn(Name::new("Alice")).id();
+        // Relations are just components, so we can add them into the bundle that we're spawning.
+        let bob = commands.spawn((Name::new("Bob"), Targeting(alice))).id();
+
+        // The `with_related` helper method on `EntityCommands` can be used to add relations in a more ergonomic way.
+        let charlie = commands
+            .spawn((Name::new("Charlie"), Targeting(bob)))
+            // The `with_related` method will automatically add the `Targeting` component to any entities spawned within the closure,
+            // targeting the entity that we're calling `with_related` on.
+            .with_related::<Targeting>(|related_spawner_commands| {
+                // We could spawn multiple entities here, and they would all target `charlie`.
+                related_spawner_commands.spawn(Name::new("Devon"));
+            })
+            .id();
+
+        // Simply inserting the `Targeting` component will automatically create and update the `TargetedBy` component on the target entity.
+        // We can do this at any point; not just when the entity is spawned.
+        commands.entity(alice).insert(Targeting(charlie));
+    }
+
+    world
+        .run_system_once(spawning_entities_with_relationships)
+        .unwrap();
+
+    fn debug_relationships(
+        // Not all of our entities are targeted by something, so we use `Option` in our query to handle this case.
+        relations_query: Query<(&Name, &Targeting, Option<&TargetedBy>)>,
+        name_query: Query<&Name>,
+    ) {
+        let mut relationships = String::new();
+
+        for (name, targeting, maybe_targeted_by) in relations_query.iter() {
+            let targeting_name = name_query.get(targeting.0).unwrap();
+            let targeted_by_string = if let Some(targeted_by) = maybe_targeted_by {
+                let mut vec_of_names = Vec::<&Name>::new();
+
+                for entity in &targeted_by.0 {
+                    let name = name_query.get(*entity).unwrap();
+                    vec_of_names.push(name);
+                }
+
+                // Convert this to a nice string for printing.
+                let vec_of_str: Vec<&str> = vec_of_names.iter().map(|name| name.as_str()).collect();
+                vec_of_str.join(", ")
+            } else {
+                "nobody".to_string()
+            };
+
+            relationships.push_str(&format!(
+                "{name} is targeting {targeting_name}, and is targeted by {targeted_by_string}\n",
+            ));
+        }
+
+        println!("{}", relationships);
+    }
+
+    world.run_system_once(debug_relationships).unwrap();
+
+    // Systems can return errors,
+    // which can be used to signal that something went wrong during the system's execution.
+    #[derive(Debug)]
+    #[expect(
+        dead_code,
+        reason = "This error struct is only used for debugging in this example."
```

Comment:
While this reason would be fine in the internals, I don't think this good when used inside an example that could be read by beginners.

Maybe something like this to explain why the expect is necessary here.
```suggestion
        reason = "Rust considers types that are only used by their debug trait as dead code."
```